com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -148,7 +237,11 @@ public GameObject GetNetworkPrefabOverride(GameObject gameObject)
         public ulong LocalClientId
         {
             get => IsServer ? NetworkConfig.NetworkTransport.ServerClientId : m_LocalClientId;
-            internal set => m_LocalClientId = value;
+            internal set
+            {
+                m_LocalClientId = value;
+                m_MessagingSystem.SetLocalClientId(value);
```

Comment:
why does `MessagingSystem` cares about the `LocalCliendId`? I guess because of loopback scenarios?

Reply:
Yeah, it just uses LocalClientId to be able to know when it's sending to itself and bypass the transport.